### PR TITLE
UT for corner case of PTXAS error before v11.1.

### PR DESCRIPTION
**Pull request by JamesTheZ**
Add a UT about PTXAS compilation error before version of 11.1. Incase we encounter the same error in a future PTXAS release.

### CODE REVIEW

File: tao_compiler/mlir/disc/tests/regression/corner_case.cc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include "tensorflow/compiler/mlir/disc/tests/mlir_feature_test.h"
+#include "tensorflow/compiler/mlir/disc/tests/mlir_test.h"
+#include "tensorflow/core/platform/test.h"
+
+namespace mlir_test {
+
+const std::string c_ft_path =
+    "tensorflow/compiler/mlir/disc/tests/regression/data/";
+
+#if CUDA_VERSION >= 11100
```
Comment (wyzero) [inline tao_compiler/mlir/disc/tests/regression/corner_case.cc:21]: Do we have this variable in in cpu-only env?

Comment (JamesTheZ) [reply tao_compiler/mlir/disc/tests/regression/corner_case.cc:21]: Then this UT will not execute because an undefined macro is replaced with 0.

Comment (linearhit) [reply tao_compiler/mlir/disc/tests/regression/corner_case.cc:21]: Should we provide a proper ptxas for users with 11.0?

Comment (JamesTheZ) [reply tao_compiler/mlir/disc/tests/regression/corner_case.cc:21]: > 

It is not the responsibility of the compiler itself. Even we provide a docker, we cannot prevent users to use other environments. (I do not think there will be that many users use our provided dockers by default.) Actually, the XLA engine will print warnings to complain about ptxas-11.0 if logging is enabled.

**Review by linearhit — APPROVED** [review]
LGTM

**Pull request merged**